src/app/app.module.spec.ts

```diff
@@ -1,10 +1,181 @@
+        walletService = new MockWalletService();
+        hockeyApp = new MockHockeyApp();
+        settingsService = new MockSettingService();
+        statusBar = new MockStatusBar();
+        platform = new MockPlatform();
+        translate = new MockTranslateService();
+
+        TestBed.configureTestingModule({
+            declarations: [SmiloWallet],
+            imports: [
+            IonicModule.forRoot(SmiloWallet),
+            TranslateModule.forRoot({
+                loader: {provide: TranslateLoader, useClass: MockTranslationLoader},
+            }),
+                ComponentsModule
+            ],
+            providers: [
+                { provide: AndroidPermissions, useValue: androidPermissions },
+                { provide: WalletService, useValue: walletService },
+                { provide: HockeyApp, useValue: hockeyApp },
+                { provide: SettingsService, useValue: settingsService },
+                { provide: StatusBar, useValue: statusBar },
+                { provide: Platform, useValue: platform },
+                { provide: TranslateService, useValue: translate }
+            ]
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(SmiloWallet);
+        comp = fixture.componentInstance;
+    });
+
+    it("should create component", () => expect(comp).toBeDefined());
 
     it("should return translatehttploader correctly", () => {
         let result = HttpLoaderFactory(null);
         expect(result).toEqual(new TranslateHttpLoader(null, "assets/i18n/"));
     });
+
+    it("should make a call to style the status bar light, prepare permissions, settings, translations, hockeyappintegration and firstpage", (done) => {
+        spyOn(comp, "ngOnInit").and.callThrough();
+        spyOn(statusBar, "styleLightContent");
+        spyOn(comp, "preparePermissions");
+        spyOn(comp, "prepareSettings");
+        spyOn(comp, "prepareHockeyAppIntegration");
+        spyOn(comp, "prepareFirstPage");
+
+        comp.ngOnInit().then(data => {
+            expect(statusBar.styleLightContent).toHaveBeenCalled();
+            expect(comp.preparePermissions).toHaveBeenCalled();
+            expect(comp.prepareSettings).toHaveBeenCalled();
+            expect(comp.prepareHockeyAppIntegration).toHaveBeenCalled();
+            expect(comp.prepareFirstPage).toHaveBeenCalled();
+            done();
+        });
+    });
+
+    it("should prepare the permissions for android", () => {
+        spyOn(platform, "is").and.callFake((arg) => {
+            if (arg === "android") return true; else return false;
+        });
+        spyOn(androidPermissions, "requestPermissions");
+
+        comp.preparePermissions();
+
+        expect(androidPermissions.requestPermissions).toHaveBeenCalledWith([androidPermissions.PERMISSION.READ_EXTERNAL_STORAGE, androidPermissions.PERMISSION.WRITE_EXTERNAL_STORAGE]);
+    });
+
+    it("should not request android permissions", () => {
```

Comment:
title should include `when not on android` to fully explain what is tested here I think.